airflow/providers/dbt/cloud/operators/dbt.py

```diff
@@ -189,15 +188,13 @@ def execute(self, context: Context):
             return self.run_id
 
     def execute_complete(self, context: Context, event: dict[str, Any]) -> int:
-        """
-        Execute when the trigger fires - returns immediately.
-
-        Relies on trigger to throw an exception, otherwise it assumes execution was successful.
-        """
-        if event["status"] == "error":
-            raise AirflowException(event["message"])
-        self.log.info(event["message"])
+        """Execute when the trigger fires - returns immediately."""
         self.run_id = event["run_id"]
+        if event["status"] == "cancelled":
+            raise DbtCloudJobRunException(f"Job run {self.run_id} has been cancelled.")
```

Comment:
Has it been discussed somewhere already as to whether we want to raise an error when the job may have deliberately been cancelled by the user?
Just unsure if we want to raise an error in such scenarios, looking for more thoughts.

Reply:
It currently works like this in the non-deferrable mode. I also think it makes sense to prevent the start of any following tasks or DAGs when the dbt job gets cancelled.

Reply:
yes, makes sense 👍🏽